subprojects/core-api/src/main/java/org/gradle/api/artifacts/transform/TransformOutputs.java

```diff
@@ -35,7 +35,31 @@
 public interface TransformOutputs {
     /**
      * Registers an output directory.
-     * For a relative path, a location for the output directory is provided.
+     *
+     * <p>
+     *     For an <strong>absolute path</strong>, the location is registered as an output directory of the {@link TransformAction}.
+     *     The path is required to point to the {@link InputArtifact}, or point inside the input artifact if it is a directory.
```

Comment:
Do you mean "the path must point to ..."?

Reply:
Fixed in https://github.com/gradle/gradle/pull/9525.